product_default_image/models/product_template.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models, tools
+
+
+class ProductTemplate(models.Model):
+
+    _inherit = 'product.template'
+
+    prevent_categ_change_image = fields.Boolean(
+        help='Used to prevent automatic changes to image fields when categ_id '
```

Comment:
Well for example, if default_get is only used, the default category on products is 'all', so the image for 'all' will load when going to create a product. However, what if I want the category to be something else and not the image for 'all' every time?

In the form I would simply change the category, but the image would not change with it. IMO that's a very poor experience, so I implemented the onchange method below.

This field prevents the image onchange from happening on products that have been created, as I found it problematic on existing products. So the image will only change with the category when creating a new product (unless the user has uploaded a custom image).

It would probably be nice to have on existing products, but I felt that was too far out of scope atm and too risky to work on.

It looks like atm the onchange is still happening on existing products in runbot, so that will need to be fixed unless you have something else in mind. Kind of weird it's True in the db but False in the onchange.